packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -686,8 +802,14 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
   function beginWork(
     current: Fiber | null,
     workInProgress: Fiber,
+    capturedValues: Array<mixed> | null,
     renderExpirationTime: ExpirationTime,
   ): Fiber | null {
+    // The effect list is no longer valid.
+    workInProgress.nextEffect = null;
+    workInProgress.firstEffect = null;
+    workInProgress.lastEffect = null;
+
```

Comment:
I see, they’re here now. Still curious why you moved.

Reply:
Not every fiber is cloned before we start work on it again (resuming), but the effect list should always be cleared.